src/App.tsx

```diff
@@ -1,61 +1,158 @@
+import React, { useState } from 'react';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import { Button, Paper, FormControl } from '@mui/material';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList';
+
+function getUser(userId: number): User | null {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  // if there is no user with a given userId
+  return foundUser || null;
+}
+
+function findUserFullname(fullName: string): User | null {
+  return usersFromServer.find(user => user.name === fullName) || null;
+}
+
+export const initialTodosList: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
+
+export const App: React.FC = () => {
+  const [userSelect, setUserSelect] = useState('');
+  const [title, setTitle] = useState('');
+  const [todos, setTodos] = useState(initialTodosList);
+  const [isErrorMessageEmptyTitle, setErrorMessageEmptyTitle] = useState(false);
+  const [isErrorMessageEmptyUser, setErrorMessageEmptyUser] = useState(false);
+
+  const handleTitleEnter = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.currentTarget.value);
+    setErrorMessageEmptyTitle(false);
+  };
+
+  const handleChooseUser = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserSelect(event.currentTarget.value);
+    setErrorMessageEmptyUser(false);
+  };
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!title || !userSelect) {
+      setErrorMessageEmptyTitle(!title);
+      setErrorMessageEmptyUser(!userSelect);
+
+      return;
+    }
```

Comment:
I think we can write this condition without a return statement.

Reply:
> I think we can write this condition without a return statement.

I think we can't, it is stopping handleSubmit execution if there is some error.

Reply:
yes, I need this return